content/5_Plat/Centroid.mdx

```diff
@@ -302,88 +314,151 @@ private static class Centroid {
 
 ## Solution - Xenia & Tree
 
+For every node, there are at most $\log N$ centroid components that include this
+node, where $N$ denotes the number of nodes. Let's call the centroid whose
+component contains one node the centroid ancestor of this node. It can also be
+shown that the path between every two nodes in the tree must include one of
+their common centroid ancestors. If we store the distance to the nearest red
+node for every centroid, we can query the minimal distance between any node and
+the nearest red node by calculating the minimum of the distance between the node
+and one of its centroid ancestors with the minimal distance from that centroid
+to a red node. To paint a node red, we just update all centroid ancestors of
+this node. Both operations can be done in $\mathcal{O} (\log N)$ time, as there
+are at most that many centroid ancestors for one node.
```

Comment:
```suggestion
For every node, there are at most $\log N$ centroid components that include this
node, where $N$ denotes the number of nodes. Let's call the centroid whose
component contains one node the centroid ancestor of this node. It can also be
shown that the path between every two nodes in the tree must include one of
their common centroid ancestors. If we store the distance to the nearest red
node for every centroid, we can query the minimal distance between any node and
the nearest red node by calculating the minimum distance between the node
and one of its centroid ancestors with the minimal distance from that centroid
to a red node. To paint a node red, we just update all centroid ancestors of
this node. Both operations can be done in $\mathcal{O} (\log N)$ time, as there
are at most that many centroid ancestors for one node.
```

> It can also be shown that the path between every two nodes in the tree must include one of their common centroid ancestors.

could you explain this briefly?